src/celengine/frame.h

```diff
@@ -23,14 +24,12 @@
  * (which specifies the coordinate axes at a given time) and the
  * nestingDepth() method (which is used to check for recursive frames.)
  */
-class ReferenceFrame
+class ReferenceFrame : public RefCounted
 {
+ protected:
```

Comment:
1. Why `protected`?
1. Why do you override empty destructor with another empty one?

Reply:
1. To prevent manual destruction. I followed original implementation.
2. Again I followed original implementation. I presumed it's because of virtuality of descructor.